text/0779-first-class-component-templates.md

```diff
@@ -0,0 +1,1613 @@
+
+This is a fairly serious developer experience problem, because it fails *invisibly* (see [this demo repository][non-colo-failure] to see this failure mode in practice).
+
+[non-colo-failure]: https://github.com/chriskrycho/fcct-interop-demo
+
+We can address in one of two ways:
+
+- Introduce support into Ember to associated non-colocated templates with their associated classes.
+- Introduce debug output which informs users that they must first migrate the referenced component to use colocation.
+
+Of these, the second option is preferable. It has significantly lower risk of introducing bugs in the framework along the way, because it only requires adding some debug alerting and does *not* require changing the semantics or implementation of long-standing Ember features. It is [straightforward to codemod][colo-codemod] to colocation.
+
+
+### The “prelude”
+
+While all values used in templates must be explicitly in scope, Ember[^glimmer-prelude] will provide some via a “prelude”.[^prelude] These are always in scope and do not need to be imported. See [RFC #0496: Handlebars Strict Mode: Keywords][rfc-0496-keyword] for a detailed list of keywords and imports.
+
+[rfc-0496-keyword]: https://emberjs.github.io/rfcs/0496-handlebars-strict-mode.html#keywords
+
+[^glimmer-prelude]: Glimmer.js may provide its own prelude. While long-term the two should likely align, this RFC simply takes the _status quo_ as a given
+
+[^prelude]: “Prelude” is the conventional name for this functionality in programming language design.
+
+
+
+### Tooling
+
+To support the new format, we need to update tooling across the ecosystem to understand the format.
+
+
+#### Syntax highlighting
+
+First, we need syntax highlighting support across the ecosystem. Support already exists [for VS Code][vsc-highlighting], which represents the single largest group of web developers; as well as for any tool which can use [a tree-sitter grammar][tree-sitter-highlighting] (e.g. Neovim).
+
+We will also need to implement support in [Linguist](https://github.com/github/linguist) for GitHub syntax highlighting.
+
+Beyond that, we should encourage the community to add support for other editors (IntelliJ, Atom, Emacs, etc.) as well as for tools like [Rouge](https://rubygems.org/gems/rouge) (which powers GitLab syntax highlighting) and other highlighters, but need not treat those as *blocking* adoption of first-class component templates.
+
+[vsc-highlighting]: https://marketplace.visualstudio.com/items?itemName=chiragpat.vscode-glimmer
+[tree-sitter-highlighting]: https://github.com/alexlafroscia/tree-sitter-glimmer
+
+
+#### Blueprints
+
+All blueprints will need to be updated to support generating the new format. During the transition period, we should allow generating both. The rollout will follow the example of the rollout of Glimmer components with Octane:
```

Comment:
Can you call out which blueprints include templates? I definitely don't think "all blueprints" use templates. I think this may be limited to components and _possibly_ tests?

Reply:
Updated; thank you for catching this!